src/useForm.tsx

```diff
@@ -50,10 +51,20 @@ export function useForm<T extends Record<string, any>>(
     return !hasValue
   }, [state])
 
+  const isDirty = useMemo(() => {
```

Comment:
Hmm "dirty" is usually used to indicate a field that has changed, but hasn't been "persisted". For example we might say that a field in memory that hasn't been saved to the DB yet is "dirty". 

But that's not really the case here. Here, you are just looking to see if the user has modified any field in the form. And so to avoid confusion, I think we'd want a different name. 

Since we already have a field-level `touched` value how about an `isTouched` or `hasBeenTouched` or something like that?